commands/operator-sdk/cmd/up/local.go

```diff
@@ -52,7 +52,9 @@ kubernetes cluster using a kubeconfig file.
 
 	upLocalCmd.Flags().StringVar(&kubeConfig, "kubeconfig", "", "The file path to kubernetes configuration file; defaults to $HOME/.kube/config")
 	upLocalCmd.Flags().StringVar(&operatorFlags, "operator-flags", "", "The flags that the operator needs. Example: \"--flag1 value1 --flag2=value2\"")
-	upLocalCmd.Flags().StringVar(&namespace, "namespace", "default", "The namespace where the operator watches for changes.")
+	// a user can set this to "" to watch all namespaces, so we need a different, invalid namespace that we can detect when
+	// deciding whether we need to get the kubeconfig's default namespace or not
+	upLocalCmd.Flags().StringVar(&namespace, "namespace", "#invalid#", "The namespace where the operator watches for changes.")
```

Comment:
My thinking on this: Would it make more sense to do `#all#` when the user wants to watch for all namespaces? `""` makes me think of blank for me not all ns.

Reply:
Kubernetes/client-go defines `""` as all namespaces, so we're just following that convention

Reply:
I believe you can use the [`cmd.Flag("namespace").Changed`](https://github.com/spf13/cobra/issues/453#issuecomment-303548563) boolean below to detect whether the user has supplied a namespace.

Reply:
@estroz Nice, didn't know about that. I'll use that instead to make this a bit cleaner